### ISSUE TITLE
re-exporting type under the same name in declare global shows error

### ISSUE THREAD

**Issue by KraXen72** [quick-lint/quick-lint-js#1173]
![image](https://github.com/quick-lint/quick-lint-js/assets/21956756/52e62296-b46b-417c-9bc7-daed5d7a8824)

i can rename it like i did with the `_VtourInfoDB`, but it's a hassle

**Comment by strager** [comment]
I am trying to replicate your issue. The following code causes TypeScript's compiler to complain about the cyclic definition:
```typescript
export type Banana = {};
declare global {
    type Banana = Banana;
}
```
[TypeScript playground](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBDAnmYcBCBDAdtjcC8cA3gL4DcAUACbADGANhlKgOb0QBGG9xFc-CZKkw4cBdLhyUSQA)

Can you share a complete example where TypeScript's compiler does not complain (but quick-lint-js does complain)?

**Comment by KraXen72** [comment]
i'll see if i can make a minimal working repro in a bit.
